src/detectors/path_traversal/path_traversal.js

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// {fact rule=path-traversal@v1.0 defects=1}
+var express = require('express')
+var app = express()
+var path = require('path')
+function pathTraversalNoncompliant() {
+    app.get('/products', (req, res) => {
+        const basePath = '/data/product/images/'
+        // Noncompliant: user-supplied path is not sanitized and could contain malicious characters.
+        var targetPath = path.join(basePath, req.query.path)
+        retrieveProduct(targetPath)
+        res.send('Here is your requested product!')
+    })
+}
+// {/fact}
+
+
+// {fact rule=path-traversal@v1.0 defects=0}
+var express = require('express')
+var app = express()
+var path = require('path')
+function pathTraversalCompliant() {
+    app.get('/products', (req, res) => {
+        const basePath = '/data/product/images/'
+        var queryPath = req.query.path
+        // Compliant: user-supplied relative path is sanitized before use.
+        queryPath = queryPath.replace(/^(\.\.(\/|\\|$))+/, '')
```

Comment:
Are we sure this replacement is strong enough to be deemed "compliant"?

There are many tricks to hack around such replacement sanitizations. For example if the sanitization is to replace `"../"` with `""`, someone could give `"..././"` to produce `"../"`.

In this case it's probably safer to either
- use an explicit allow-list, or
- restrict the user-supplied path component to a single word that uses only letters and no symbols at all, like `[a-z]+`.

Reply:
Improved the sanitisation to check a single word with no symbols.